hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,111 @@
+        self.test_particle_type = self._simulation.state._cpp_sys_def.getParticleData().getTypeByName(self.test_particle_type)
+
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be an HPMC integrator.")
+
+        integrator_pairs = None
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            integrator_pairs = [(integrate.Sphere, _hpmc.ComputeFreeVolumeSphere),
+                                (integrate.ConvexPolygon, _hpmc.ComputeFreeVolumeConvexPolygon),
+                                (integrate.SimplePolygon, _hpmc.ComputeFreeVolumeSimplePolygon),
+                                (integrate.ConvexPolyhedron, _hpmc.ComputeFreeVolumeConvexPolyhedron),
+                                (integrate.ConvexSpheropolyhedron, _hpmc.ComputeFreeVolumeSpheropolyhedron),
+                                (integrate.Polyhedron, _hpmc.ComputeFreeVolumePolyhedron),
+                                (integrate.Ellipsoid, _hpmc.ComputeFreeVolumeEllipsoid),
+                                (integrate.ConvexSpheropolygon, _hpmc.ComputeFreeVolumeSpheropolygon),
+                                (integrate.FacetedEllipsoid, _hpmc.ComputeFreeVolumeFacetedEllipsoid),
+                                (integrate.Sphinx, _hpmc.ComputeFreeVolumeSphinx),
+                                (integrate.SphereUnion, _hpmc.ComputeFreeVolumeSphereUnion),
+                                (integrate.ConvexSpheropolyhedronUnion, _hpmc.ComputeFreeVolumeConvexPolyhedronUnion),
+                                (integrate.FacetedEllipsoidUnion, _hpmc.ComputeFreeVolumeFacetedEllipsoidUnion)]
+        else:
+            integrator_pairs = [(integrate.Sphere, _hpmc.ComputeFreeVolumeSphereGPU),
+                                (integrate.ConvexPolygon, _hpmc.ComputeFreeVolumeConvexPolygonGPU),
+                                (integrate.SimplePolygon, _hpmc.ComputeFreeVolumeSimplePolygonGPU),
+                                (integrate.ConvexPolyhedron, _hpmc.ComputeFreeVolumeConvexPolyhedronGPU),
+                                (integrate.ConvexSpheropolyhedron, _hpmc.ComputeFreeVolumeSpheropolyhedronGPU),
+                                (integrate.Polyhedron, _hpmc.ComputeFreeVolumePolyhedronGPU),
+                                (integrate.Ellipsoid, _hpmc.ComputeFreeVolumeEllipsoidGPU),
+                                (integrate.ConvexSpheropolygon, _hpmc.ComputeFreeVolumeSpheropolygonGPU),
+                                (integrate.FacetedEllipsoid, _hpmc.ComputeFreeVolumeFacetedEllipsoidGPU),
+                                (integrate.SphereUnion, _hpmc.ComputeFreeVolumeSphereUnionGPU),
+                                (integrate.ConvexSpheropolyhedronUnion, _hpmc.ComputeFreeVolumeConvexPolyhedronUnionGPU),
+                                (integrate.FacetedEllipsoidUnion, _hpmc.ComputeFreeVolumeFacetedEllipsoidUnionGPU)]
```

Comment:
Use getattr and string concatenation to simplify this code. This is what another PR uses in a similar way:
```
                my_class = getattr(_md, 'TwoStepRATTLENVE' + self.manifold_constraint.name)
            else:
                my_class = getattr(_md, 'TwoStepRATTLENVE' + self.manifold_constraint.name + 'GPU')
```